### ISSUE TITLE
IEC support for SD2IEC works unreliably in Mega65 mode

### ISSUE THREAD

**Issue by sir-lazarus** [MEGA65/mega65-rom-public#71]
Platform MEGA65, ROM 920377 and newer:

When I use my SD2IEC device on my MEGA65's IEC port in Mega65 mode, it works unreliably.
At first glance it works. But loading is very slow and when I load larger size files it doesn't stop loading.
Everything is slow as expected when switching to 64 mode, but works fine. So no issue in 64 Mode.

Everything works fine with the C64 Core, especially with Jiffydos.

The issues are only in Mega65 mode.

**Comment by dansanderson** [comment]
MEGA65 mode has IEC timing issues across the board, unfortunately. Printers and non-1581 drives have issues from MEGA65 mode. It's one of the last remaining unfinished areas of the ROM.

**Comment by sir-lazarus** [comment]
Yes I know.
When this topic is on the bench, support the JiffyDOS protocol would be awesome, because SD2IEC supports this.

**Comment by eilers** [comment]
I have a problem with my 1581 Replica. This Replica uses original chips and a (afaik) correct replication of the original 1581 mother board. I would expect that it behaves equally as an original 1581 from commodore. But I cannot prove that as I don't have an original device for comparison.

## Problem description (versions see below):
The drives works without problems using the  C64-core as well as within the C64-Mode on the Mega65-Core. 

When I use a recent MEGA65 ROM and for instance use the `DIR` command, the green light on the drive lights up but nothing happens. The drive is not spinning up or anything else happens. Only the `DIR` command ist blocking the Mega65.
When I use the original prototype ROM for the C65, the behavior is different: The `DIR` command does not stuck and shows the directory. But the end of the directory is crumbled. Loading of files with `DLOAD` seems to work, but was not extensively tested.

## Versions:
C64-Core: WiP-V5.1-RC2
Mega65-Core: 0.96 3c10488
Mega65 ROM: 920404
C65 ROM: 0.9B.910828
Mega65-Board: R3

## Picture of the Board of the 1581:
Please note: I replaced the 6502 on the picture with one from an old 1541. The replaced 6502 is from 1984. The one on this  picture was definitely a fake as the numbers make no sense.

![IMG_0761](https://github.com/MEGA65/mega65-rom-public/assets/115845/8f650bcb-9b4b-4182-a039-90d0d7529886)

**Comment by dansanderson** [comment]
I have a vintage 1581 where `DIR` stalls with v0.96 and succeeds with the HW IEC branch, so that's a positive sign. We have some bugs to work out, but it validates the approach. Also the JiffyDOS implementation appears to be doing its job with my SD2IEC. (I don't believe I have any other JiffyDOS enabled drives, but I can borrow some from friends if needed.)

**Comment by eilers** [comment]
Yes. I can confirm that the HW IEC implementation solves the problem. 
I have a slightly outdated test matrix here that shows that there is only one issue left with my SD2IEC on the R6 board. The R3 Board looks fine.

See: https://wiki.eilers-online.net/IEC-Core-Test
